## Task 5.1 Reads in two csv files (the 2019 and 2020 BPD crime data), concatenates them, then drops duplicates and rows in which null values appear in certain columns. This dataframe is then saved to a new csv

In [1]:
import pandas as pd

bcd19 = pd.read_csv("bcd19.csv", low_memory=False)
bcd20 = pd.read_csv("bcd20.csv", low_memory=False)

bcd_merge = pd.concat([bcd19, bcd20])
bcd_merge.drop_duplicates(keep='first') #unclear from instructions if meant to drop duplicates only if they occur in the below subset of columns or in any column
bcd_merge.dropna(subset=['INCIDENT_NUMBER', 'OFFENSE_CODE', 'DISTRICT', 'Long', 'Lat', 'SHOOTING', 'OCCURRED_ON_DATE'])
pd.DataFrame.to_csv(bcd_merge, "bcd19-20.csv")

## Task 5.2 Reads in the newly created csv file, then provides a frequency count of crime incidents organized by grouping by year and month, then doing a count

In [2]:
bcd_complete = pd.read_csv("bcd19-20.csv", low_memory=False)
bcd_complete.groupby(['YEAR','MONTH'])['INCIDENT_NUMBER'].count()

YEAR  MONTH
2019  1        6958
      2        6173
      3        7093
      4        6963
      5        7783
      6        7635
      7        8083
      8        8430
      9        8184
      10       7121
      11       6468
      12       6293
2020  1        6342
      2        5897
      3        5513
      4        4387
      5        5439
      6        5925
      7        6236
      8        6533
      9        6484
      10       6766
      11       5770
      12       5602
Name: INCIDENT_NUMBER, dtype: int64

[editted]

In [3]:
[editted]

Property Crimes  Violent Crimes
YEAR                                 
2019            13186            8307
2020            12259            5182

### Interpretation: For both property crimes and violent crimes, incident counts declined between 2019 and 2020 

## Task 5.4 Counts the total number of violent vs property crime incidents in each year by district. It follows the procedure of 5.3, but drops null and 'External' values found in the 'DISTRICT' column, before grouping by District then Year

In [4]:
bcd_pcount2 = bcd_complete[pd.DataFrame(bcd_complete['OFFENSE_CODE'].tolist()).isin(p_crimes).any(1).values].dropna(subset=['DISTRICT'])
bcd_pcount2 = bcd_pcount2[bcd_pcount2.DISTRICT != 'External'].groupby(['DISTRICT', 'YEAR'])['INCIDENT_NUMBER'].count()
bcd_vcount2 = bcd_complete[pd.DataFrame(bcd_complete['OFFENSE_CODE'].tolist()).isin(v_crimes).any(1).values].dropna(subset=['DISTRICT'])
bcd_vcount2 = bcd_vcount2[bcd_vcount2.DISTRICT != 'External'].groupby(['DISTRICT', 'YEAR'])['INCIDENT_NUMBER'].count()

freq_pv2 = pd.concat([bcd_pcount2,bcd_vcount2],axis=1)
freq_pv2.columns = ['Property Crimes','Violent Crimes']
freq_pv2

Property Crimes  Violent Crimes
DISTRICT YEAR                                 
A1       2019             1910            1208
         2020             1501             732
A15      2019              269             152
         2020              337              87
A7       2019              416             329
         2020              512             247
B2       2019             1404            1263
         2020             1415             825
B3       2019             1073            1071
         2020              887             595
C11      2019             1651            1007
         2020             1423             629
C6       2019              956             618
         2020             1132             406
D14      2019              880             419
         2020              960             244
D4       2019             2779            1129
         2020             2536             755
E13      2019              687             428
         2020              569             278
E18      2019              592             345
         2020              476             191
E5       2019              434             249
         2020              459             160

### Interpretation: While crime seemed to decline within districts, from 2019 to 2020, there still remains significant variation in the incident counts across district, ranging from a low of 87 in District A15, during 2020, to a high of 2779 in District D4, during 2019.